cli/azd/pkg/project/framework_service_npm.go

```diff
@@ -109,8 +109,7 @@ func (np *npmProject) Package(
 				return
 			}
 
-			// Run Build, injecting env.
-			envs := append(np.env.Environ(), "NODE_ENV=production")
+			envs := []string{"NODE_ENV=production"}
```

Comment:
Does this means that azd builds `production` for node all the times?
Even if NODE_ENV is set in customer's environment, is azd overriding it with this value to build?

I know that's not part of this PR, but I got scared looking at this...

Reply:
@vhvb1989 Let me also think about this more. I'm investigating our environment usage in builds further to illustrate the less-ideal behaviors and see how we can approach these problems.

Reply:
I am confused about why we are setting/injecting this from azd.
I wonder if this is because we want to emulate what react-apps-cli does (https://create-react-app.dev/docs/adding-custom-environment-variables) automatically with NODE_ENV.  But react-apps also takes care of replacing the env var for the value during build time (for static apps).
But, for other node apps (like a backend node server), calling build does not replace env vars, so I am not sure what would it be used for.

Reply:
I arrive separately at the same conclusion. Let's remove it.